subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -492,11 +492,78 @@ The example above centrally defines plugin versions in the settings script's plu
 Note that alternatively you could also move the version declarations in the root project `plugins {}` block.
 
 
-// [[sec:collections]]
-// == TODO Working with collections
-// overview & api
-// === Via Kotlin delegated properties
-// === Container scope string invoke extension
+[[sec:containers]]
+== Working with containers
+
+The Gradle build model makes heavy use of containers.
+For example, Gradle core contributes `configurations` which is a container of `Configuration` objects and `tasks` which is a container of `Task` objects ; core plugins such as `java` or `distribution` contribute containers such as `sourceSets` or `distributions`.
+Community plugins also contribute containers, such as `android.buildTypes` by the `android` plugin.
+Gradle plugins create containers using `Project.container<Type>()`.
+
+[TIP]
+--
+Note that if you are configuring existing elements on supported containers you can leverage the type-safe accessors as described in the <<sec:configuring_plugins>> section above.
+--
+
+The Gradle Kotlin DSL provides with uniform ways of interacting with containers.
```

Comment:
:o: IANANS but _provides with_ without an object reads strange, maybe _provides **build authors** with uniform ways of interacting with containers._?